Update Social Media Auth page with Twitter Signin (#157)
invertase/react-native-firebase-docs: docs/auth/social-auth.md

```

## Twitter

> You can use react-native-twitter-signin for Twitter authentication. This library handles the flow of logging in a user and accesssing their `authToken` and `authTokenSecrent` by wrapping around the official Twitter library.

Reviewer: ```suggestion
You can use `react-native-twitter-signin` for Twitter authentication. This library handles the flow of logging in a user and accessing their `authToken` and `authTokenSecret` by wrapping around the official Twitter library.
```

The`react-native-twitter-signin` library allows us to login (using `RNTwitterSignIn`) which returns `authToken` and `authTokenSecret`

> Make sure you thorougly follow the instructions on setting up the necessary requirements to get react-native-twitter-signin working.

Reviewer: ```suggestion
Make sure you thoroughly follow the instructions on setting up the necessary requirements to get `react-native-twitter-signin` working.
```

```js

> import {NativeModules} from 'react-native}

Reviewer: ```suggestion
import {NativeModules} from 'react-native}';
```

Author: There's a random bracket at the end, it should be

import { NativeModules } from 'react-native';

Reviewer: Updated

import firebase from 'react-native-firebase';

const {RNTwitterSignIn} = NativeModules;

> const TwittrKeys = {

Reviewer: ```suggestion
const TwitterKeys = {
```

>    TWITTER_CONSUMER_KEY: 'PLACE_YOUR_TWITTER_CONSUMER_KEY_HERE'

Reviewer: ```suggestion
   TWITTER_CONSUMER_KEY: 'PLACE_YOUR_TWITTER_CONSUMER_KEY_HERE',
```

   TWITTER_CONSUMER_SECRET: 'PLACE_YOUR_TWITTER_CONSUMER_SECRET_HERE'

> }

Reviewer: ```suggestion
};
```

export const signinTwitter = async = () => { 

>     RNTwitterSignin.init(TwitterKeys.TWITTER_CONSUMER_KEY, TwitterKeys.TWITTER_CONSUMER_SECRET);

Reviewer: ```suggestion
    RNTwitterSignIn.init(TwitterKeys.TWITTER_CONSUMER_KEY, TwitterKeys.TWITTER_CONSUMER_SECRET);
```

    
    try {

>     const result = await RNTwitterSignIn.logIn();

Reviewer: ```suggestion
      const result = await RNTwitterSignIn.logIn();
```

>     const credential = firebase.auth.TwitterAuthProvider.credential(result.authToken, result.authTokenSecret);

Reviewer: ```suggestion
      const credential = firebase.auth.TwitterAuthProvider.credential(result.authToken, result.authTokenSecret);
```

>     await firebase.auth().signInWithCredential(credential);

Reviewer: ```suggestion
      await firebase.auth().signInWithCredential(credential);
```
